llvm/include/llvm/SYCLLowerIR/CleanupSYCLCompilerInternalMetadata.h

```diff
@@ -0,0 +1,30 @@
+//===-------------- CleanupSYCLCompilerInternalMetadata.h ---------------------
+//---------------- CleanupSYCLCompilerInternalMetadata Pass ---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Cleanup SYCL compiler internal metadata inserted by the frontend and used by
+// sycl-post-link as it will never be used in the compilation ever again
```

Comment:
Once again: removing metadata in `BackendUtils` makes it disappear _before_ `sycl-post-link`